dotCMS/src/main/java/com/dotcms/ai/app/AIModels.java

```diff
@@ -154,24 +152,31 @@ public void resetModels(final Host host) {
      * @return a list of supported model names
      */
     public List<String> getOrPullSupportedModels() {
-        final List<String> cached = supportedModelsCache.getIfPresent(SUPPORTED_MODELS_KEY);
-        if (CollectionUtils.isNotEmpty(cached)) {
-            return cached;
+        synchronized (supportedModelsCache) {
```

Comment:
why are you using a sync block here, if need a safe collection use a ConcurrentHashMap